src/puppetlabs/trapperkeeper/authorization/ring.clj

```diff
@@ -38,13 +38,13 @@
    certificate :- (schema/maybe X509Certificate)]
   (assoc-in request [:authorization :certificate] certificate))
 
-(schema/defn authorized-name :- schema/Str
+(schema/defn authorized-name :- (schema/maybe schema/Str)
   "Get the authorized client name from the request."
   [request :- Request]
-  (get-in request [:authorization :name] ""))
+  (get-in request [:authorization :name]))
 
 (schema/defn set-authorized-name :- Request
   "Set the authorized client name onto the request."
   [request :- Request
    name :- (schema/maybe schema/Str)]
-  (assoc-in request [:authorization :name] (str name)))
+  (assoc-in request [:authorization :name] name))
```

Comment:
This is probably okay.  This value is consumed by Puppet Server upstream but it appears that it was tolerating a `nil` properly before anyway.  Generally speaking, I like to have my APIs that externally expose a String value to return an empty string instead of a nil - just to reduce the likelihood that the client might blow up if it isn't properly checking for nil before dereferencing.  If it were straightforward to keep this as an empty string, that would be my preference.  But I'm okay going with nil here.  @dankreek - WDYT?